config/routes.rb

```diff
@@ -6,6 +6,8 @@
   resources :periods
   resources :reports
 
+  get "reports/:id/hide" => "reports#hide", as: :hide_report
```

Comment:
You could also [nest this route](https://guides.rubyonrails.org/routing.html#adding-member-routes) under the `resources :reports` routes, to get something like

```suggestion
  resources :reports do 
    get "/hide", as: :hide_report, on: :member
   end
```